File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        
   
	int a = m-1, b = n-1;
	int k = m + n - 1;

	while(a >= 0 && b >= 0){
		if(nums1[a] > nums2[b]){
			nums1[k] = nums1[a];
			a -= 1;
			k -= 1;
		}
		else if (nums2[b] > nums1[a]){
			nums1[k] = nums2[b];
			b -= 1;
			k -= 1;
		}
		else {
			nums1[k] = nums2[b];
			b -= 1;
			k -= 1;
		}
	}

	while(a >= 0){
		nums1[k] = nums1[a];
		a -= 1;
		k -= 1;
	}

	while(b >= 0){
		nums1[k] = nums2[b];
		b -= 1;
		k -= 1;
	}
     
        
        
        
    }
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        l, r = m - 1, n - 1
        k = m + n - 1
        while l >= 0 and r >= 0:
            if nums2[r] >= nums1[l]:
                nums1[k] = nums2[r]
                r -= 1
                k -= 1
            else:
                nums1[k] = nums1[l]
                l -= 1
                k -= 1

        while l >= 0:
            nums1[k] = nums1[l]
            l -= 1
            k -= 1

        while r >= 0:
            nums1[k] = nums2[r]
            r -= 1
            k -= 1


        